src/cli.rs

```diff
@@ -89,6 +91,13 @@ pub struct Cli {
     #[structopt(short = "s", long = "style", possible_values = &Theme::variants(), case_insensitive = true)]
     pub theme: Option<Theme>,
 
+    /// Proxy to use for a specific protocol. The value passed to this option should take the form of
+    /// <PROTOCOL>:<PROXY_URL>. You can specify proxies for multiple protocols by repeating this option.
+    /// Note that when this option is absent, the environment variables `http_proxy` and
+    /// `https_proxy` can be used to set proxies to use.
```

Comment:
Can you add an example value? HTTPie has `http:http://foo.bar:3128`, but maybe something where the traffic's protocol and the proxy's own protocol don't match would be clearer. (It confused me at first.)

Reply:
What do you think about my latest change? I'm not quite sure if making the output of `--help` much longer is useful, but I agree that the values you should pass to this option aren't very intuitive.

Reply:
I think the amount of information is good. It helps to reorder it slightly and insert blank lines, so everything most people need to know is in the first or second section. After revising it a little I have this:
```
    /// Proxy to use for a specific protocol. The value passed to this option should take the
    /// form of PROTOCOL:PROXY_URL. For example: `--proxy https:http://proxy.host:8080`.
    ///
    /// PROTOCOL can be `http`, `https`, or `all`.
    ///
    /// If your proxy requires credentials, put them in the URL, like so:
    /// `--proxy http:socks5://user:password@proxy.host:8000`.
    ///
    /// You can specify proxies for multiple protocols by repeating this option.
    ///
    /// The environment variables `$http_proxy` and `$https_proxy` can also be used.
```
HTTPie's help text is often pretty wordy.

Reply:
I like it. I had originally used blank lines, but decided to remove them, as _clap_ then decides on a different way to render the long help. Here's what it looks like now when using your suggestion:

```
xh 0.7.0

USAGE:
    xh [FLAGS] [OPTIONS] <[METHOD] URL> [--] [REQUEST_ITEM]...

FLAGS:
        --offline         
            Construct HTTP requests without sending them anywhere

    -j, --json            
            (default) Data items from the command line are serialized as a JSON object

    -f, --form            
            Data items from the command line are serialized as form fields

    -m, --multipart       
            Similar to --form, but always sends a multipart/form-data request (i.e., even without files)

    -I, --ignore-stdin    
            Do not attempt to read stdin

    -F, --follow          
            Do follow redirects

    -d, --download        
            Download the body to a file instead of printing it

    -h, --headers         
            Print only the response headers, shortcut for --print=h

    -b, --body            
            Print only the response body, Shortcut for --print=b

    -c, --continue        
            Resume an interrupted download

    -v, --verbose         
            Print the whole request as well as the response

    -q, --quiet           
            Do not print to stdout or stderr

    -S, --stream          
            Always stream the response body

        --check-status    
            Exit with an error status code if the server replies with an error

        --help            
            Prints help information

    -V, --version         
            Prints version information


OPTIONS:
    -A, --auth-type <auth-type>              
            Specify the auth mechanism [possible values: Basic, Bearer]

    -a, --auth <USER[:PASS] | TOKEN>         
            

    -o, --output <output>                    
            Save output to FILE instead of stdout

        --max-redirects <max-redirects>      
            Number of redirects to follow, only respected if `follow` is set

    -p, --print <print>                      
            String specifying what the output should contain

        --pretty <pretty>                    
            Controls output processing [possible values: All, Colors, Format, None]

    -s, --style <theme>                      
            Output coloring style [possible values: Auto, Solarized]

        --proxy <PROTOCOL:PROXY_URL>...      
            Proxy to use for a specific protocol. The value passed to this option should take the form of
            <PROTOCOL>:<PROXY_URL>. For example: `--proxy https:http://proxy.host:8080`.
            
            PROTOCOL can be `http`, `https` or `all`.
            
            If your proxy requires credentials, put them in the URL, like so: `--proxy
            http:socks5://user:password@proxy.host:8000`.
            
            You can specify proxies for multiple protocols by repeating this option.
            
            When omitting this option, the environment variables `http_proxy` and `https_proxy` can also be used.
        --default-scheme <default-scheme>    
            The default scheme to use if not specified in the URL


ARGS:
    <[METHOD] URL>       
            The request URL, preceded by an optional HTTP method

    <REQUEST_ITEM>...    
            Optional key-value pairs to be included in the request
```

I don't mind, per se, but it was a choice I wasn't comfortable making.

Reply:
Ah, I hadn't realized that. That might actually be a problem. Especially now that `-h` is short for `--header` instead of printing a shortened `--help`.

For now I guess I'd tend toward something compressed, then:
```rust
    /// Use a proxy for a protocol. For example: `--proxy https:http://proxy.host:8080`.
```
In the long term we should look for a better solution. Maybe we can convince `--help` to print the short help, and move the long help to `--long-help` or a man page (or both?) @ducaale, do you have any ideas?

Reply:
>In the long term we should look for a better solution. Maybe we can convince --help to print the short help, and move the long help to --long-help or a man page (or both?)

Sounds good to me. I will create an issue for this as a reminder.